src/engine.c

```diff
@@ -4126,6 +4129,14 @@ int settle_needed(game *g, int who, int which, int special[], int num_special,
 				if (!(o_ptr->code & P3_ALIEN) &&
 				    good == GOOD_ALIEN) return -1;
 
+				/* Check for correct chromo */
```

Comment:
I have not tested the case. I have not worked out how the debugging system works, neither put much thought to the problem.